File: Random/Arrays & Hashing/group_anagrams.js
var groupAnagrams = function(strs) {
    let map = {}
    for(let str of strs){
        let reversedStr = str.split('').sort().join('');
        if(map[reversedStr]){
            map[reversedStr].push(str);
        }else{
            map[reversedStr] = [str];
        }
    }
    return Object.values(map);
    
};


groupAnagrams(["eat","tea","tan","ate","nat","bat"])

File: Random/Arrays & Hashing/group_anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dict_ = {}
        for str in strs:
            sorted_str = "".join(sorted(list(str)))
            if sorted_str not in dict_:
                dict_[sorted_str] = [str]
            else:
                dict_[sorted_str].append(str)
        return list(dict_.values())


s = Solution()
print(s.groupAnagrams(["eat","tea","tan","ate","nat","bat"]))